### PR TITLE
 Initial Database Schema for Image and Model Tracking

### PR DESCRIPTION
**Pull request by Elshaday-Tamire**
This PR introduces a simplified and scalable database schema to support the core components of our wildfire detection pipeline. The structure is designed to be future-proof and manageable by the partner team post-handoff.

---

### 🖼️ `images` table

| Field           | Type      | Description |
|----------------|-----------|-------------|
| `id`           | `String` (UUID) | Unique image ID |
| `image_path`   | `String`  | Local path to image file |
| `label_path`   | `String`  | Local path to label JSON |
| `classes`      | `JSON`    | Array of class labels (e.g., `["Fire", "Smoke"]`) |
| `bboxes`       | `JSON`    | Array of bounding boxes (e.g., `[[x1, y1, x2, y2], ...]`) |
| `label_status` | `Enum`    | One of `auto`, `human`, or `mismatch` |
| `semi_supervised` | `JSON` | Optional additional info from pre-labeling |
| `tags`         | `JSON`    | Array of custom tags (e.g., `["augmented", "smoke"]`) |

---

### 🧠 `models` table

| Field             | Type   | Description |
|------------------|--------|-------------|
| `id`             | `String` (UUID) | Model ID |
| `model_path`     | `String` | Path to model weights file |
| `model_type`     | `Enum`   | One of `base`, `distilled`, or `quantized` |
| `model_status`   | `Enum`   | `training` or `trained` |
| `train_image_ids`| `JSON`   | List of image UUIDs used for training |
| `val_image_ids`  | `JSON`   | List of image UUIDs used for validation |
| `test_image_ids` | `JSON`   | List of image UUIDs used for testing |
| `training_history` | `JSON` | Dict with training metrics, loss, etc. |

---

### 🧪 `experiments` table

| Field         | Type    | Description |
|---------------|---------|-------------|
| `id`         | `String` (UUID) | Experiment ID |
| `model_id`   | `String` | FK to `models.id` |
| `augmentation` | `JSON` | JSON config for augmentations (e.g., `{"flip": true}`) |
| `random_seed` | `String` | Seed used for reproducibility |
| `performance` | `JSON` | Evaluation metrics (e.g., `{"mAP": 0.89}`) |
| `config`      | `JSON` | Training settings (e.g., `{"batch_size": 16}`) |

---

## ⚙️ .env Configuration

Create a `.env` file in the root of your project:

```env
# For SQLite (default)
DATABASE_URL=sqlite:///db.sqlite3

# Or for PostgreSQL
# DATABASE_URL=postgresql://username:password@localhost:5432/your_database
```

---

## 🖥️ Connecting via TablePlus (SQLite)

1. Open TablePlus and click `Create a New Connection`.
2. Choose **SQLite** as the database type.
3. Under `Path`, select the `db.sqlite3` file in your project root.
4. Click **Connect**.
image
5. You’ll now be able to inspect `images`, `models`, and `experiments` tables.
image

### CODE REVIEW

**Comment by sep-he** [comment]
Looks good. Has the components we discussed with the with the partner.

**Review by Arc-Celt — APPROVED** [review]
Nicely done!
